src/main/navigation/navigation.c

```diff
@@ -655,12 +658,14 @@ static navigationFSMEvent_t navOnEnteringState_NAV_STATE_ALTHOLD_INITIALIZE(navi
         resetAltitudeController(navTerrainFollowingRequested());
     }
 
-    if (((prevFlags & NAV_CTL_ALT) == 0) || ((prevFlags & NAV_AUTO_RTH) != 0) || ((prevFlags & NAV_AUTO_WP) != 0) || terrainFollowingToggled) {
+    if ((motorStatus == MOTOR_STOPPED_USER) || ((prevFlags & NAV_CTL_ALT) == 0) || ((prevFlags & NAV_AUTO_RTH) != 0) || ((prevFlags & NAV_AUTO_WP) != 0) || terrainFollowingToggled) {
         setupAltitudeController();
         setDesiredPosition(&navGetCurrentActualPositionAndVelocity()->pos, posControl.actualState.yaw, NAV_POS_UPDATE_Z);  // This will reset surface offset
     }
 
-    return NAV_FSM_EVENT_SUCCESS;
+    if (motorStatus != MOTOR_STOPPED_USER) return NAV_FSM_EVENT_SUCCESS; // user asked to restart the motor, resume ALTHOLD
```

Comment:
I believe when we exit the "pause" we should reset the altitude setpoint - hence `setDesiredPosition` should be called here, before switching to "in-progress" state